taichi/backends/vulkan/vulkan_program.cpp

```diff
@@ -68,6 +85,9 @@ void VulkanProgramImpl::materialize_runtime(MemoryPool *memory_pool,
   *result_buffer_ptr = (uint64 *)memory_pool->allocate(
       sizeof(uint64) * taichi_result_buffer_entries, 8);
 
+// TODO: Support for Android if possible... Should we move this code to be the
+// responsability of the caller to this library?
```

Comment:
When embedded in other applications the design is to leave the creation of VkDevice and related states to the caller/host. VulkanDeviceCreator is only used when Taichi is running on its own.

Reply:
I think it's good for now!